GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -430,15 +433,7 @@ public Point GetQuickItemSelectorLocation()
 
         private void ResetNavigationHistory()
         {
-            var selectedRevisions = GetSelectedRevisions();
-            if (selectedRevisions.Count == 1)
-            {
-                _navigationHistory.Push(selectedRevisions[0].ObjectId);
-            }
-            else
-            {
-                _navigationHistory.Clear();
-            }
+            _navigationHistory.Clear();
```

Comment:
❓ I have not analyzed how it did / should work, and have not run it yet. At first glance, the following might be what it should (have) be(en):
```
            _navigationHistory.Clear();
            if (selectedRevisions.Count == 1)
            {
                _navigationHistory.Push(selectedRevisions[0].ObjectId);
            }
```

Reply:
This happened to work before as there were no selected revisions yet when this was running. So the history was always reset. (The risk for a race condition was minimal.)
This call is changed now to reset when HEAD is changed so the navigation history contains commits in the grid.
I realize right now that the handling is not working when a filter is applied, the filter must always be refreshed (or the reset is changed to validate or so and is reset when the history contains commits that is not in the grid).
(#9833 would create popups when the history is invalid.)

I have not used the history so often: Shift-Alt-Left (or right)